packages/theme/src/cli/utilities/theme-fs.ts

```diff
@@ -144,27 +144,26 @@ export function mountThemeFileSystem(root: string, options?: ThemeFileSystemOpti
       return file.value || file.attachment || ''
     })
 
-    const syncPromise = contentPromise.then(async (content) => {
-      if (!unsyncedFileKeys.has(fileKey)) return false
+    const syncPromise = contentPromise
+      .then(async (content) => {
+        if (!unsyncedFileKeys.has(fileKey)) return false
 
-      const [result] = await bulkUploadThemeAssets(Number(themeId), [{key: fileKey, value: content}], adminSession)
+        const [result] = await bulkUploadThemeAssets(Number(themeId), [{key: fileKey, value: content}], adminSession)
+
+        if (!result?.success) {
```

Comment:
(nit) Would prefer `result.errors` here

Reply:
I'm not familiar with the underlying API this calls in `cli-kit`, so not sure if something can have a "warning" in errors or something like that. [It looks like](https://github.com/Shopify/cli/blob/main/packages/cli-kit/src/public/node/themes/factories.ts#L71-L72) `success` is just `response.status === 200`, and errors are at least `{}` instead of undefined.
cc @jamesmengo thoughts?